demosplan/DemosPlanCoreBundle/Logic/Procedure/ProcedureTypeService.php

```diff
@@ -363,16 +374,10 @@ public function addMissingRequestData(
 
                 return $return;
             })->toArray();
-            $params[$formName]['statementFormDefinition']['fieldDefinitions'] = $fieldDefinitions;
+            $statementFormDefinition['fieldDefinitions'] = $fieldDefinitions;
+            $request->request->set('statementFormDefinition', $statementFormDefinition);
         }
 
-        // Adds id to the form if missing
-        if (!isset($params[$formName]['id'])) {
-            $params[$formName]['id'] = $params['id'];
-        }
-
-        $request->request->add([$formName => $params[$formName]]);
```

Comment:
Is it correct that you completely delete this general add?

Reply:
yes, this is intended, because we are adding/setting the needed values directly to the request instead of creating and filling '$params'